routing/pathfind.go

```diff
@@ -196,13 +196,14 @@ func (s sortableRoutes) Swap(i, j int) {
 
 // newRoute returns a fully valid route between the source and target that's
 // capable of supporting a payment of `amtToSend` after fees are fully
-// computed. If the route is too long, or the selected path cannot support the
-// fully payment including fees, then a non-nil error is returned.
+// computed. If the route is too long, the selected path cannot support the
+// fully payment including fees, or the path requires more than *maxFee fees,
+// then a non-nil error is returned.
 //
 // NOTE: The passed slice of ChannelHops MUST be sorted in forward order: from
 // the source to the target node of the path finding attempt.
 func newRoute(amtToSend lnwire.MilliSatoshi, pathEdges []*ChannelHop,
-	currentHeight uint32) (*Route, error) {
+	currentHeight uint32, maxFee *lnwire.MilliSatoshi) (*Route, error) {
```

Comment:
To avoid using a nilable parameter, you could make it non-nilable, and then when you extract the `maxFee` you set `maxFee = payAmt` if no user specified max :)

Reply:
I thought about that, but I wasn't sure if it would be right to set a hard limit at `amt`. Would we want it to be impossible for the user to pay a large fee if they really need/want to? E.g. Alice wants to send 1 satoshi to Carol ASAP, but the only route between them requires a 2 satoshi fee. Should Alice never be able to make such a payment?

Reply:
Good point! What about setting it to some very large number in that case instead?

Reply:
Sounds good! How about `math.MaxInt64`?

Reply:
Maybe you could use the constant `rpcserver.maxPaymentMSat`. Cannot imagine anybody want to pay a higher fee than the max allowed payment amount :P

Reply:
Perfect, thanks!

Reply:
@halseth I ended up bringing back the nilable parameter. The reason for this is that roasbeef requested that I make the MaxFee field of the LightningPayment struct optional. Besides at extraction, this function seems to be the most convenient place to check for `nil` on that field. I could certainly move that functionality up to router.SendPayment, but I believe the outcome would be the same. Let me know what you think!